### PR TITLE
Migrate Auth from JWT to OAuth

### PR DESCRIPTION
**Pull request by enavitaniuc**
Change Authentication from JWT to OAuth (JWT will soon stop working)

## Description

- altered usage of `aepp.configure`, swapped `technical_account_id` and `private_key_path` with `scope` this triggers OAuth2 Auth execution flow.
- updated README

## Motivation and Context

The Service Account (JWT) credentials have been deprecated in favor of the OAuth Server-to-Server credentials. The applications using the Service Account (JWT) credentials will stop working after Jan 1, 2025. 

## How Has This Been Tested?

- tested notebooks locally

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

NOTE: there seem to be some formatting changes caused by newer version of jupyter

### CODE REVIEW

File: notebooks/assignments/Week4Notebook.ipynb [hunk 1]
```diff
@@ -3125,7 +3118,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.10.2"
+   "version": "3.12.2"
```
Comment (gjamuar) [inline notebooks/assignments/Week4Notebook.ipynb:3121]: I see different python version in `toml` file. Are all the notebooks tested on this latest python version, if yes then lets update the whole project's python. Right now notebooks have different version than project python version.

**Review by gjamuar — APPROVED** [review]
Approved with minor comment

File: notebooks/assignments/Week4Notebook.ipynb (hunk 1 above)
Comment (enavitaniuc) [reply notebooks/assignments/Week4Notebook.ipynb:3121]: from what i understand `toml` - enforces minimal version to 3.9; i have nothing against bumping it to 3.12 min, additionally i refrained to commit changes caused by 3.12 runs since the noise causes PR to explode and makes it difficult to figure to track relevant changes.

File: pyproject.toml [hunk 2]
```diff
@@ -6,7 +6,7 @@ authors = ["cmle dev team"]
 readme = "README.md"
 
 [tool.poetry.dependencies]
-python = "^3.9"
+python = "^3.12"
```
Comment (cmenguy) [inline pyproject.toml:9]: Why is that necessary?

**Review by cmenguy — APPROVED** [review]
LGTM thanks for tackling this 👍

File: pyproject.toml (hunk 2 above)
Comment (enavitaniuc) [reply pyproject.toml:9]: testing was performed on version 3.12- as part of first roun dof review @gjamuar recommending bumping python version so we have same version listed in the toml and in notebooks;
